src/nvim/ops.c

```diff
@@ -2524,6 +2528,46 @@ static void yank_copy_line(yankreg_T *reg, struct block_def *bd, long y_idx)
   *pnew = NUL;
 }
 
+static void yank_do_autocmd(oparg_T *oap, yankreg_T *reg)
+{
+  char buf[NUMBUFLEN+2];
+
+  if (!has_event(EVENT_TEXTYANKPOST)) {
+    // No autocommand was defined
+    return;
+  }
+
+  // set v:event to a dictionary with information about the yank
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
+
+  // the yanked text
+  list_T *list = list_alloc();
+  for (linenr_T i = 0; i < reg->y_size; i++) {
+    list_append_string(list, reg->y_array[i], -1);
+  }
+  list->lv_lock = VAR_FIXED;
+  dict_add_list(dict, "regcontents", list);
```

Comment:
Hmm, `set_vim_var_dict` sets them (I mean inside the dict, not the assignment to the `v:` dict, with `=` though `|=` would be better because of the ALLOC as you said) and I tested doing the same and
`au! TextYankPost * let v:event.regcontents = 0`
and it errors out as it should.